### ISSUE TITLE
Partial GNOME 46 support

### ISSUE THREAD

**Issue by cedricroijakkers** [vinceliuice/Matcha-gtk-theme#230]
The GNOME 46 update dropped. So far I noticed two things that seem broken:

Notification popups:

![image](https://github.com/vinceliuice/Matcha-gtk-theme/assets/15158042/82580552-e911-4373-b54f-b93e2c8b9eaa)

Colours in the calendar:

![image](https://github.com/vinceliuice/Matcha-gtk-theme/assets/15158042/9eb5a6e4-6d20-4157-b0ee-72f45bffa987)

**Comment by xuanruiqi** [comment]
Yeah, could @vinceliuice possibly apply the same fixes that applies to the other themes - hopefully this will fix the problem?

**Comment by mandibleman** [comment]
i would also appreciate it if the ~~border: none;~~ fix could be implemented here

**Comment by vinceliuice** [comment]
Fixed now

**Comment by superuser-miguel** [comment]
![Screenshot from 2024-04-30 16-45-23](https://github.com/vinceliuice/Matcha-gtk-theme/assets/16271056/92079ce0-7975-442d-bc5c-ddb7674fd997)
![Screenshot from 2024-04-27 16-44-47](https://github.com/vinceliuice/Matcha-gtk-theme/assets/16271056/75cc4701-ff5f-44bd-b905-b104047ab7d7)

Still broken. I have tried using the CSS with Gradience. What else can I try ?

**Comment by RaulRohjans** [comment]
> ![Screenshot from 2024-04-30 16-45-23](https://private-user-images.githubusercontent.com/16271056/326977233-92079ce0-7975-442d-bc5c-ddb7674fd997.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ1MTQzOTcsIm5iZiI6MTcxNDUxNDA5NywicGF0aCI6Ii8xNjI3MTA1Ni8zMjY5NzcyMzMtOTIwNzljZTAtNzk3NS00NDJkLWJjNWMtZGRiNzY3NGZkOTk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDMwVDIxNTQ1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyZjA3OWRhZWIxMDkzZmFlZGM0NTczNzRlZTBkODJiNjdmNzU2ODlkZDcyYjFiODhmNzNkNGMxMWVmMTExOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-NXcwmfPnwHJHCBliasJXgoxFH2SgbfLk-26h0hkm50) ![2024-04-27_19-57-54](https://private-user-images.githubusercontent.com/16271056/326977660-ef35d69f-179b-4aea-93f0-cb0d7ec16639.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ1MTQzOTcsIm5iZiI6MTcxNDUxNDA5NywicGF0aCI6Ii8xNjI3MTA1Ni8zMjY5Nzc2NjAtZWYzNWQ2OWYtMTc5Yi00YWVhLTkzZjAtY2IwZDdlYzE2NjM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDMwVDIxNTQ1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2NmFkYjM3ZDExZDg0MjIzYmM2MTBjYzE2YjY3ZDkyMWQxNjk4NTM1YjVmODk0Y2I3MDI3MDNmYjJmZWExMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KPIya3QxEt_6_c83PEJPEgKiDseCfbBu02KCF4kytdc)
> 
> Still broken. I have tried using the CSS with Gradience. What else can I try ?

You can try replacing the content of your ~/.config/gtk-3.0 with the gtk-3.0 folder contents of the repository, and do the same for the ~/.config/gtk-4.0, replacing it with the content of gtk-4.0.

Make a backup of those folder before doing this!

Afterwards, try a reboot and it should work fine :)

**Comment by superuser-miguel** [comment]
@RaulRohjans While it has fixed some apps that didn't work prior, Notifications are completely broken. Any other things to try that come to mind?

**Comment by RaulRohjans** [comment]
@superuser-miguel Not a lot, no. Have you tried completely removing the theme's files and reinstalling from source? The issue is fixed, it seems your system is reading an older version of your theme install

**Comment by superuser-miguel** [comment]
@RaulRohjans I'll have to give that a try now. I get a lot of notifications so it's really a bother at the moment.

**Comment by RaulRohjans** [comment]
@salazarbarrera This one, Matcha-gtk-theme the folder paths are the following:

`/src/gtk/gtk-3.0
/src/gtk/gtk-4.0`

You copy the file contents from those into your user folders respectively.

`~/.config/gtk-3.0
~/.config/gtk-4.0`

**Comment by salazarbarrera** [comment]
Oooh, copying into, I thought you meant replace the whole directory. Since the contents are so different. I'm on my phone now, but I'll try it later.

![imagen](https://github.com/vinceliuice/Matcha-gtk-theme/assets/68821455/7f0110ac-a16d-448e-889b-97a8281af3bb)

**Comment by RaulRohjans** [comment]
@superuser-miguel  @salazarbarrera 
I noticed I made a big mistake, its not the contents of the repo folder you have to copy, sorry!

You want to install the theme first through 

`./install`

Then copy the contents of `~/.themes/<MatchaVariantYouWant>/gtk-3.0` to `~/.config/gtk-3.0` and the same for gtk-4.0, so from `~/.themes/<MatchaVariantYouWant>/gtk-4.0` to `~/.config/gtk-4.0`

Again, so sorry for the confusion. :(

**Comment by salazarbarrera** [comment]
ooooh I got you. Unfortunately it doesn't work. For the record, I already did something similar before trying this. I linked the files from one directory to the other as it was instructed in the output of the package manager

> To force Libadwaita appliations to use the gtk-4.0 theme,
> symlink the gtk-4.0 theme to config folder
> (replace Matcha-dark-sea with chosen theme variant):
> 
> mkdir -p ~/.config/gtk-4.0/
> ln -sf /usr/share/themes/Matcha-dark-sea/gtk-4.0/{assets,gtk.css,gtk-dark.css} ~/.config/gtk-4.0/
>  

But copying the files as you suggested didn't work either.

**Comment by superuser-miguel** [comment]
For now I am back to how it was when I sent the first message as the responses and recommendations did not work. I do see some new information here, so i will try those. I reverted in part because it did nothing for notifications and various elements were broken afterwards.

I'll report back with updates later.

**Comment by frost19k** [comment]
Erm... guys, the install script does this automatically. 

```bash
  -l, --libadwaita         Force all libadwaita app use linked gtk-4.0 theme
```

**Comment by salazarbarrera** [comment]
Tangentially related to the issue:
Manjaro recently sent a stable update package for [GNOME Shell](https://archlinux.org/packages/extra/x86_64/gnome-shell/),  going from 1:45.4-1 to the 1:46.1-1 version of the Arch package (I think it's irrelevant, but [gnome-tweaks](https://archlinux.org/packages/extra/x86_64/gnome-tweaks/) also updated from 45.1-1 to 46.1-1 accordingly). After this, the notifications problem was fixed:
![image](https://github.com/vinceliuice/Matcha-gtk-theme/assets/68821455/c5f34b92-8b60-45ca-ac96-17120cd318b7)

On the opposite side, downgrading Matcha to its 20231030-2 Arch package (the one which worked with GNOME 45), causes a similar bug in GNOME 46.
![image](https://github.com/vinceliuice/Matcha-gtk-theme/assets/68821455/987d01ca-15b9-485d-b65f-56bf9fc58f38)

In summary, and based _**only**_ in what I've experienced with my machine with this distro:

- The current 20240501-1 matcha-gtk-theme release works with GNOME 46.1, but shows unreadable giant notifications in 45.4.
- The previous 20231030-2 matcha-gtk-theme release works with GNOME 45.4, but shows unreadable giant notifications in 46.1.

So, for some reason (which I don't know because I have never themed GNOME by myself), the last version of the theme is not backwards compatible with GNOME 45, and GNOME 46 doesn't support the previous version. So, my recommendation would be that everyone sticks with the old Matcha version until their distro decides to upgrade GNOME.

**Comment by karolyi** [comment]
... Manjaro with gnome 46 will break lots of stuff anyway :) As it did for me.

Yes, the only viable workaround is to stick with the old version, there is a downgrade tool for Manjaro.

**Comment by superuser-miguel** [comment]
Finally got around to fixing my setup. Notifications work properly now. Basically : 
* Moved the contents of both GTK3/4 directories to `~/.config/gtk-X.0 ` and ran the install script.

**Comment by frost19k** [comment]
@superuser-miguel I've had that config since Gnome 42. Notifications are still broken for me on GNOME 46.2

```bash
❯ lds ~/.config | grep gtk
lrwxrwxrwx  1 maazb maazb   41 May 15 10:20 gtk-3.0 -> /usr/share/themes/Matcha-dark-sea/gtk-3.0
lrwxrwxrwx  1 maazb maazb   41 May 15 10:20 gtk-4.0 -> /usr/share/themes/Matcha-dark-sea/gtk-4.0

❯ git rev-parse --short HEAD
4f1de198 
```

**Comment by frost19k** [comment]
I believe that the cause of the notification issue is the GDM theme. If you install the GTK theme without the GDM theme then notifications seem fine.
